src/test/java/de/tum/cit/aet/artemis/lecture/ExerciseUnitIntegrationTest.java

```diff
@@ -144,16 +144,16 @@ void createExerciseUnit_asStudent_shouldForbidden() throws Exception {
         ExerciseUnit exerciseUnit = new ExerciseUnit();
         exerciseUnit.setExercise(exercise);
         exerciseUnit.setId(1L);
-        request.postWithResponseBody("/api/lecture/lectures/" + lecture1.getId() + "/exercise-units", exerciseUnit, ExerciseUnit.class, HttpStatus.BAD_REQUEST);
+        request.postWithResponseBody("/api/lecture/lectures/" + lecture1.getId() + "/exercise-units", exerciseUnit, ExerciseUnit.class, HttpStatus.FORBIDDEN);
     }
 
     @Test
     @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
-    void createExerciseUnit_notExistingLectureId_shouldReturnNotFound() throws Exception {
+    void createExerciseUnit_notExistingLectureId_shouldReturnForbidden() throws Exception {
         Exercise exercise = course1.getExercises().stream().findFirst().orElseThrow();
         ExerciseUnit exerciseUnit = new ExerciseUnit();
         exerciseUnit.setExercise(exercise);
-        request.postWithResponseBody("/api/lecture/lectures/" + 0 + "/exercise-units", exerciseUnit, ExerciseUnit.class, HttpStatus.NOT_FOUND);
+        request.postWithResponseBody("/api/lecture/lectures/" + 0 + "/exercise-units", exerciseUnit, ExerciseUnit.class, HttpStatus.FORBIDDEN);
     }
```

Comment:
I verified my assumption by changing the PreAuthorizations in `ExerciseUnitResource` back (without the `inLecture` part) - we then receive the bad request and not found again 

There is also a wrong test user used in this file for (should be a TA, as editors have access)
```
    @Test
    @WithMockUser(username = TEST_PREFIX + "instructor42", roles = "INSTRUCTOR")
    void createExerciseUnit_notInstructorInCourse_shouldReturnForbidden() throws Exception {
```